src/internal/storage/fileset/merge.go

```diff
@@ -49,23 +49,21 @@ func (mr *MergeReader) iterate(ctx context.Context, cb func(File) error) error {
 	return pq.Iterate(func(ss []stream.Stream) error {
 		var fss []*fileStream
 		for _, s := range ss {
-			fss = append(fss, s.(*fileStream))
+			fs := s.(*fileStream)
+			if fs.deletive {
+				fss = nil
+				continue
+			}
+			fss = append(fss, fs)
+		}
+		if len(fss) == 0 {
+			return nil
 		}
 		if len(fss) == 1 {
-			if fss[0].deletive {
-				return nil
-			}
 			return cb(newFileReader(ctx, mr.chunks, fss[0].file.Index()))
```

Comment:
I'm confused to why this case wasn't being hit previously

Reply:
Files normally get into the system through put file which creates a delete and add operation for the files. When we are merging here, this will result with two file streams (one for the delete and one for the add). In this case, and really any case where there is only one additive stream, we don't need to use a merge file reader since we are not merging content from multiple file sets.